googler

```diff
@@ -1175,13 +1206,23 @@ class GoogleParser(html.parser.HTMLParser):
             self.set_handlers_to('result')
             return 'result'
 
+        # Autocorrect
+        if tag == 'span' and 'spell_orig' in self.classes(attrs) and attrs.get('id') != 'sifm':
+            self.autocorrected = True
+            return
+        if tag == 'a' and 'spell' in self.classes(attrs) and attrs.get('id') != 'srfl':
```

Comment:
> just trying to keep it low if we can.

I'm afraid we can't optimize here. There's a bit of nickle and diming you can do in the conditions at the cost of overall consistency in the code base. Not much you can do otherwise in a linear search for a bunch of things in rather hostile, poorly structured input.

To truly improve the sensitivity to one added condition, what you can do is scan the tree once, cache the results in sort of search tree(s), then select with logarithmic (or better) complexity. It would be super hardcore though, and whether it will actually save time is questionable for what we do. I checked bs4 source code and it's simply doing a linear traversal for finding and selector queries; see [here](https://git.launchpad.net/~vivekchandola/beautifulsoup/+git/trunk/tree/library/bs4/element.py#n1296) and [here](https://git.launchpad.net/~vivekchandola/beautifulsoup/+git/trunk/tree/library/bs4/element.py#n1518). I don't know how JavaScript folks implement CSS selector queries (`querySelectorAll` and simple special cases).

Anyway, this consideration is not very helpful, since processing time is dwarfed by I/O anyway.